model_api/python/model_api/models/visual_prompting.py

```diff
@@ -0,0 +1,757 @@
+"""
+ Copyright (C) 2024 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from collections import defaultdict
+from itertools import product
+from typing import Any, NamedTuple
+
+import cv2
+import numpy as np
+from model_api.models import SAMDecoder, SAMImageEncoder
+from model_api.models.utils import VisualPromptingResult
+
+
+class VisualPromptingFeatures(NamedTuple):
+    feature_vectors: np.ndarray
+    used_indices: np.ndarray
+
+
+class Prompt(NamedTuple):
+    data: list[np.ndarray] | np.ndarray
+    labels: list[np.ndarray | int] | np.ndarray
```

Comment:
`Prompt` class seems to contain data related to multiple prompts. I think it's cleaner to have it be a single prompt and then use a list. Something like:
```python
class Prompt(NamedTuple):
    data: np.ndarray
    label: int

...

def infer(
    self,
    image: np.ndarray,
    boxes: list[Prompt] | None = None,
    points: list[Prompt] | None = None,
) -> VisualPromptingResult:
```

Reply:
Done